File: cpp/Pembeli.cpp
#include <iostream>
using namespace std;

class Pembeli {
    private: // atribut private
        int idPembeli; // unique identifier untuk pembeli
        string namaPembeli; // nama pembeli
        string nomerTelepon; // nomer telepon pembeli
        string alamat; // alamat pembeli
    public:
        // =========== Constructor =========== //
        Pembeli(int idPembeli, string namaPembeli, string alamat, string nomerTelepon) {
            this->idPembeli = idPembeli;
            this->namaPembeli = namaPembeli;
            this->nomerTelepon = nomerTelepon;
            this->alamat = alamat;
        }

        //============ Setter ============//
        // method untuk mengatur id pembeli
        void setIdPembeli(int idPembeli) {
            this->idPembeli = idPembeli; // menggunakan this-> untuk membedakan antara atribut kelas dan parameter fungsi
        }
        
        // method untuk mengatur nama pembeli
        void setNamaPembeli(string namaPembeli) {
            this->namaPembeli = namaPembeli;
        }

        // method untuk mengatur nomer telepon pembeli
        void setNomerTelepon(string nomerTelepon) {
            this->nomerTelepon = nomerTelepon;
        }

        // method untuk mengatur alamat pembeli
        void setAlamat(string alamat) {
            this->alamat = alamat;
        }



        //============ Getter ============//
        int getIdPembeli() {
            return this->idPembeli;
        }

        // method untuk mendapatkan nama pembeli
        string getNamaPembeli() {
            return this->namaPembeli;
        }

        // method untuk mendapatkan nomer telepon pembeli
        string getNomerTelepon() {
            return this->nomerTelepon;
        }

        // method untuk mendapatkan alamat pembeli
        string getAlamat() {
            return this->alamat;
        }

        // method untuk menampilkan informasi pembeli
        void displayInfo() {
            cout << "+----------------------------------+\n";
            cout << "|          DATA PEMBELI            |\n";
            cout << "+----------------------------------+\n";
            cout << "| ID     : " << idPembeli << "\n";
            cout << "| Nama   : " << namaPembeli << "\n";
            cout << "| Telepon: " << nomerTelepon << "\n";
            cout << "| Alamat : " << alamat << "\n";
            cout << "+----------------------------------+\n\n";
        }


        // =========== Destructor =========== //
        ~Pembeli() {
        }
};

File: java/Pembeli.java
public class Pembeli {
    // =========== Atribut Private =========== //
    private int idPembeli;       // unique identifier untuk pembeli
    private String namaPembeli;  // nama pembeli
    private String nomerTelepon; // nomor telepon pembeli
    private String alamat;       // alamat pembeli

    // =========== Constructor =========== //
    public Pembeli(int idPembeli, String namaPembeli, String alamat, String nomerTelepon) {
        this.idPembeli = idPembeli;
        this.namaPembeli = namaPembeli;
        this.nomerTelepon = nomerTelepon;
        this.alamat = alamat;
    }

    // =========== Setter =========== //
    // Setter untuk mengubah nilai atribut private
    public void setIdPembeli(int idPembeli) {
        this.idPembeli = idPembeli;
    }

    // Setter untuk mengubah nilai atribut private
    public void setNamaPembeli(String namaPembeli) {
        this.namaPembeli = namaPembeli;
    }

    // Setter untuk mengubah nilai atribut private
    public void setNomerTelepon(String nomerTelepon) {
        this.nomerTelepon = nomerTelepon;
    }

    // Setter untuk mengubah nilai atribut private
    public void setAlamat(String alamat) {
        this.alamat = alamat;
    }

    // =========== Getter =========== //
    // Getter untuk mendapatkan nilai atribut private
    public int getIdPembeli() {
        return this.idPembeli;
    }

    // Getter untuk mendapatkan nilai atribut private
    public String getNamaPembeli() {
        return this.namaPembeli;
    }

    // Getter untuk mendapatkan nilai atribut private
    public String getNomerTelepon() {
        return this.nomerTelepon;
    }

    // Getter untuk mendapatkan nilai atribut private
    public String getAlamat() {
        return this.alamat;
    }

    // =========== Method untuk menampilkan info =========== //
    public void displayInfo() {
        System.out.println("+----------------------------------+");
        System.out.println("|          DATA PEMBELI            |");
        System.out.println("+----------------------------------+");
        System.out.println("| ID     : " + idPembeli);
        System.out.println("| Nama   : " + namaPembeli);
        System.out.println("| Telepon: " + nomerTelepon);
        System.out.println("| Alamat : " + alamat);
        System.out.println("+----------------------------------+\n");
    }
}
